### PR TITLE
Add `.pick` and `.omit`

### PR DESCRIPTION
**Pull request by aryaemami59**
This PR:

  - [X] Adds `.pick` and `.omit` methods which work similarly to `.extract` and `.exclude`.

### CODE REVIEW

File: test/usage.test.ts [hunk 1]
```diff
@@ -401,3 +401,15 @@ test('Another limitation: passing `this` references to `expectTypeOf` results in
...
+  type Person = {name: string; age: number}
+
+  expectTypeOf<Person>().pick<'name'>().toEqualTypeOf<{name: string}>()
+})
+
+test('Use `.omit` to remove a set of properties from an object', () => {
+  type Person = {name: string; age: number}
+
+  expectTypeOf<Person>().omit<'name'>().toEqualTypeOf<{age: number}>()
+})
```
Comment (mmkal) [inline test/usage.test.ts:415]: Could you move these up to come after the extract/exclude tests? And then run `pnpm lint --fix` - that will update the readme too.

**Review by mmkal — CHANGES_REQUESTED** [review]
Thanks for this! Looks good, just need to generate the docs (eslint should do automatically based on the tests)

File: test/usage.test.ts (hunk 1 above)
Comment (aryaemami59) [reply test/usage.test.ts:415]: Sure thing, I will do that!

File: README.md
Comment (aryaemami59) [inline README.md:1]: I believe this whole file changed due to line ending issues.

Comment (mmkal) [reply README.md:1]: Yeah, maybe I have something misconfigured in gitattributes. Are you on windows, out of curiosity?

Comment (aryaemami59) [reply README.md:1]: I'm on windows.

**Comment by aryaemami59** [comment]
@mmkal  looks like snapshots needed to be updated.

**Review by mmkal — APPROVED** [review]

**Pull request merged**